tests/ui/const-generics/wrong-normalization.rs

```diff
@@ -0,0 +1,18 @@
+// This test ensures that if implementation on projections is supported,
+// it doesn't end in very weird cycle error.
+
+#![crate_type = "lib"]
+
+pub trait Identity {
+    type Identity: ?Sized;
+}
+
+impl<T: ?Sized> Identity for T {
+    type Identity = Self;
+}
+
+pub struct I8<const F: i8>;
+
+impl <I8<{i8::MIN}> as Identity>::Identity { //~ ERROR
```

Comment:
Can you include the error message here, instead of just `//~ ERROR`?

Reply:
Sure. Although it's not really relevant here.